### PR TITLE
Add qPI, qKG, qUCB, qMultiStepHEntropySearch acquisition functions

### PR DESCRIPTION
**Pull request by martinakaduc**

### CODE REVIEW

File: sequence_design/acqfs.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        y = (
+            self.model.sample(
+                torch.tensor(X),
+                sample_size=64,
+            )
+            .mean(0)
+            .squeeze(-1)
+            .float()
+            .detach()
+            .cpu()
```
Comment (sangttruong) [inline sequence_design/acqfs.py:127]: Why do we send the acqf value to CPU here? @martinakaduc

Comment (martinakaduc) [reply sequence_design/acqfs.py:127]: The acquisition functions are implemented under reward models. We will host a reward server for training PPO. Thus after computing acquisition values as reward scores, we should send it to CPU for reward server to send them to PPO process..

**Review by sangttruong — APPROVED** [review]
